### PR TITLE
Bot code improvement (2/??)

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

My last PR was Beepsky code improvements, I said I'd continue this on if it was well received, and it seemed to be, so I'm back again.
This time I'm not doing another bot, but instead I'm doing bots directly.

- Removed unused vars
- Removed a timer var for a define instead
- replaced 1/0 with TRUE/FALSE
- re-organized the list of vars to fit where they should be, since all newer vars were just thrown at the bottom
- Autodocs all vars
- Merges text_hack/text_dehack/text_dehack_fail into just 'hackables', which is what you will hack/emag, rather than having lines of text be vars.
- Because of above, I was able to remove a now-useless set_custom_texts proc
- Replaces a bunch of if else with early returns
- ALL bots have a 50% chance of dropping their robot arm now, instead of being snowflaked into nearly every individual bot.
- Merges a ton of vars into mode_flags and cover_flags, to handle things ranging from if it can have a pAI installed, to whether the bot is emagged.
- Changes bot types from flags to defines, and removes var/model as we can now instead use bot_type's define.
- Fixes Secbots not dropping their batons by reverting the var merge I made in last PR

I'll continue this when I can. I hope to re-organize the vars on all subtype bots in order of which they appear where they are defined.

## Why It's Good For The Game

I'm hoping this would make bot code less shitty and easier to make things for, it feels unreadable to newer people to me.

## Changelog
:cl:
fix: Bots will no longer turn invisible when being turned on/off
/:cl:

### CODE REVIEW

**Comment by jjpark-kb** [comment]
So since this PR is here... your previous PR made the armsky variant, could you perhaps make an officer beepsky variant as well? There are some maps that have var-edited beepskys with a different name and var (that is shared between them).

**Comment by JohnFulpWillard** [comment]
> 
> 
> So since this PR is here... your previous PR made the armsky variant, could you perhaps make an officer beepsky variant as well? There are some maps that have var-edited beepskys with a different name and var (that is shared between them).
> 
> EDIT: Yea, it's on Delta, Icebox, Kilo, and Meta image

Sure

**Comment by jjpark-kb** [comment]
> > So since this PR is here... your previous PR made the armsky variant, could you perhaps make an officer beepsky variant as well? There are some maps that have var-edited beepskys with a different name and var (that is shared between them).
> > EDIT: Yea, it's on Delta, Icebox, Kilo, and Meta image
> 
> Sure
> 
> Edit: It seems the default beepsky that it's varediting, is meant to be the actual beepsky. I have no idea why the maps are varediting this instead of just leaving it be the default.

thank you!

**Review by Maurukas — APPROVED** [review]
Mapping is good. 

I can't tell how much of the code changes are readability vs refactor, so someone competent should probably look at that part.

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 1]
```diff
@@ -20,19 +20,16 @@
...
 	radio_key = /obj/item/encryptionkey/headset_med
 	radio_channel = RADIO_CHANNEL_MEDICAL
-
 	bot_type = MED_BOT
-	model = "Medibot"
-	bot_core_type = /obj/machinery/bot_core/medbot
-	window_id = "automed"
-	window_name = "Automatic Medical Unit v1.1"
 	data_hud_type = DATA_HUD_MEDICAL_ADVANCED
+	hackables = "healt processor circuits"
```
Comment (Kylerace) [inline code/modules/mob/living/simple_animal/bot/medbot.dm:30]: healt?

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 2]
```diff
@@ -22,93 +23,123 @@
...
-	var/shuffle = FALSE // If we should shuffle our adjacency checking
-
-	var/blockcount = 0 //number of times retried a blocked path
-	var/awaiting_beacon = 0 // count of pticks awaiting a beacon response
-
-	var/nearest_beacon // the nearest beacon's tag
-	var/turf/nearest_beacon_loc // the nearest beacon's location
+
+	///Bot-related mode flags on the Bot indicating how they will act.
+	var/bot_mode_flags = BOT_MODE_ON | BOT_MODE_REMOTE_ENABLED | BOT_MODE_PAI_CONTROLLABLE
```
Comment (Kylerace) [inline code/modules/mob/living/simple_animal/bot/bot.dm:53]: why are you moving so many of these important vars into bitflags?

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/bot/bot.dm (hunk 2 above)
Comment (JohnFulpWillard) [reply code/modules/mob/living/simple_animal/bot/bot.dm:53]: I thought it was better to have 1 flag than 5 different vars for things that all have some common theme.

**Comment by LemonInTheDark** [comment]
@JohnFulpWillard you need to update your changelog with 🆑 tags so the parser actually reads it

**Comment by JohnFulpWillard** [comment]
> @JohnFulpWillard you need to update your changelog with 🆑 tags so the parser actually reads it

hahhaha

**Pull request merged**